webui/react/src/pages/Dashboard.tsx

```diff
@@ -0,0 +1,253 @@
+} from 'components/Table/Table';
+import useModalJupyterLab from 'hooks/useModal/JupyterLab/useModalJupyterLab';
+import { paths } from 'routes/utils';
+import {
+  getCommands,
+  getExperiments,
+  getJupyterLabs,
+  getProjectsByUserActivity,
+  getShells,
+  getTensorBoards,
+} from 'services/api';
+import Icon from 'shared/components/Icon/Icon';
+import Spinner from 'shared/components/Spinner';
+import usePolling from 'shared/hooks/usePolling';
+import { dateTimeStringSorter } from 'shared/utils/sort';
+import { useCurrentUser } from 'stores/users';
+import { ShirtSize } from 'themes';
+import { CommandTask, DetailedUser, ExperimentItem, Project } from 'types';
+import { Loadable } from 'utils/loadable';
+
+import css from './Dashboard.module.scss';
+
+const FETCH_LIMIT = 25;
+const DISPLAY_LIMIT = 25;
+
+const Dashboard: React.FC = () => {
+  const [experiments, setExperiments] = useState<ExperimentItem[]>([]);
+  const [tasks, setTasks] = useState<CommandTask[]>([]);
+  const [projects, setProjects] = useState<Project[]>([]);
+  const [submissions, setSubmissions] = useState<Submission[]>([]);
+  const [canceler] = useState(new AbortController());
+  const [tableLoading, setTableLoading] = useState<boolean>(true);
+  const [projectsLoading, setProjectsLoading] = useState<boolean>(true);
+  const loadableCurrentUser = useCurrentUser();
+  const currentUser = Loadable.match(loadableCurrentUser, {
+    Loaded: (cUser) => cUser,
+    NotLoaded: () => undefined,
+  });
+  const { contextHolder: modalJupyterLabContextHolder, modalOpen: openJupyterLabModal } =
+    useModalJupyterLab();
+  type Submission = ExperimentItem & CommandTask;
+
+  const fetchTasks = useCallback(
+    async (user: DetailedUser) => {
+      const [commands, jupyterLabs, shells, tensorboards] = await Promise.all([
+        getCommands({
+          limit: FETCH_LIMIT,
+          orderBy: 'ORDER_BY_DESC',
+          signal: canceler.signal,
+          sortBy: 'SORT_BY_START_TIME',
+          users: [user.id.toString()],
+        }),
+        getJupyterLabs({
+          limit: FETCH_LIMIT,
+          orderBy: 'ORDER_BY_DESC',
+          signal: canceler.signal,
+          sortBy: 'SORT_BY_START_TIME',
+          users: [user.id.toString()],
+        }),
+        getShells({
+          limit: FETCH_LIMIT,
+          orderBy: 'ORDER_BY_DESC',
+          signal: canceler.signal,
+          sortBy: 'SORT_BY_START_TIME',
+          users: [user.id.toString()],
+        }),
+        getTensorBoards({
+          limit: FETCH_LIMIT,
+          orderBy: 'ORDER_BY_DESC',
+          signal: canceler.signal,
+          sortBy: 'SORT_BY_START_TIME',
+          users: [user.id.toString()],
+        }),
+      ]);
+      const newTasks = [...commands, ...jupyterLabs, ...shells, ...tensorboards];
+      setTasks(newTasks);
+    },
+    [canceler],
+  );
+
+  const fetchExperiments = useCallback(
+    async (user: DetailedUser) => {
+      const response = await getExperiments(
```

Comment:
just realized that there's no API error handling within `fetch***` funcs

Reply:
thanks, added!